### PR TITLE
svc: Add env variables for max/default service accounts expiration

### PR DESCRIPTION
**Pull request by vadmeste**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/common-main.go [hunk 1]
```diff
@@ -802,6 +802,34 @@ func serverHandleEnvVars() {
...
+				err = fmt.Errorf("value less than %s", minServiceAccountExpiry)
+			case *envVar.val > maxServiceAccountExpiry:
+				err = fmt.Errorf("value greater than %s", maxServiceAccountExpiry)
+			}
+		}
+		logger.FatalIf(err, "Unable to parse "+envVar.name)
+	}
+
+	if globalEnvMaxSVCTTL > 0 && globalEnvDefaultSVCTTL > globalEnvMaxSVCTTL {
+		logger.Fatal(errors.New("TTL cannot be greated than MINIO_IAM_SERVICE_ACCOUNTS_MAX_TTL"), "Unexpected MINIO_IAM_SERVICE_ACCOUNTS_DEFAULT_TTL value")
```
Comment (shtripat) [inline cmd/common-main.go:831]: `%s/greated/greater/g`

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Who is asking for this ?

**Comment by harshavardhana** [comment]
@vadmeste While this is okay for the most part, we cannot standardize this kind of approach.  Adding new ENVs for every new feature will not work, and it's hard to document it cleanly.

https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_iam-condition-keys.html We should go via this kind of approach where we can indicate that

```
svc:DurationSeconds
```

To be conceptually allowed as per policy this way, we can permanently restrict this per user via a central policy.

**Comment by harshavardhana** [comment]
Closing this PR since https://github.com/minio/minio/pull/18835 is already sent

**Pull request closed without merging**